internal/operator/backrest/repo.go

```diff
@@ -235,6 +237,32 @@ func getRepoDeploymentFields(clientset kubernetes.Interface, cluster *crv1.Pgclu
 	return &repoFields
 }
 
+// UpdateAnnotations updates the annotations in the "template" portion of a
+// pgBackRest deployment
+func UpdateAnnotations(clientset kubernetes.Interface, cluster *crv1.Pgcluster,
+	annotations map[string]string) error {
+	// get a list of all of the instance deployments for the cluster
+	deployment, err := operator.GetBackrestDeployment(clientset, cluster)
+
+	if err != nil {
+		return err
+	}
+
+	// now update the pgBackRest deployment
+	log.Debugf("update annotations on [%s]", deployment.Name)
+	log.Debugf("new annotations: %v", annotations)
+
+	deployment.Spec.Template.ObjectMeta.SetAnnotations(annotations)
+
+	// finally, update the Deployment. If something errors, we'll log that there
+	// was an error, but continue with processing the other deployments
+	if _, err := clientset.AppsV1().Deployments(deployment.Namespace).Update(deployment); err != nil {
```

Comment:
In general been thinking we prob want to start accounting for potential conflicts when using `Update()` (i.e. `PUT` requests), e.g. in case something else updates the Deployment underneath us.  Or the alternative is to use a patch instead.

Def not a blocker or anything (we have similar Updates all over the code base), but something I wanted to at least mention since it's been in the back of my mind with some recent changes (I've personally be leaning on patch a bit more).

Reply:
Sounds like a good topic as we continue to evolve the Operator codebase.

Reply:
I have similar thoughts about patching. As is, this reads well and has boundaries that look good, but I also know we should try to write entire objects less often.

As an example, here's one way to replace the annotations using a JSON Patch:

```go
patch, err := json.Marshal([]util.JSONPatchOperation{{
  Op: "add",
  Path: "/spec/template/metadata/annotations",
  Value: annotations,
}})

if err == nil {
  _, err = clientset.AppsV1().Deployments(cluster.Namespace).Patch(deploymentName, types.JSONPatchType, patch)
}

return err
```